soc/arm/atmel_sam/common/atmel_sam_dt.h

```diff
@@ -36,13 +36,24 @@
 #define ATMEL_SAM_PIN_PERIPH(inst, i) \
 	DT_PHA(NODE_ID_FROM_PINCTRL_0(inst, i), atmel_pins, peripheral)
 
+/* Helper function for ATMEL_SAM_PIN_FLAGS */
+#define ATMEL_SAM_PIN_FLAG(inst, i, flag) \
+	DT_PROP(NODE_ID_FROM_PINCTRL_0(inst, i), flag)
+
+/* Convert DT flags to SoC flags */
+#define ATMEL_SAM_PIN_FLAGS(inst, i) \
+	ATMEL_SAM_PIN_FLAG(inst, i, bias_pull_up) << SOC_GPIO_PULLUP_POS | \
```

Comment:
Could you put the RHS into parentheses rather than requiring it be done at point-of-use?

Reply:
Done.